File: api/schemas/match.py
from typing import Optional

from pydantic import BaseModel

from api.schemas.competitor import Competitor


class Match(BaseModel):
    id: int
    ordinal: int
    competitor_1: Optional[Competitor]
    competitor_1_score: Optional[int]
    competitor_2: Optional[Competitor]
    competitor_2_score: Optional[int]
    next_match_id: Optional[int]
    status: int

    class Config:
        orm_mode = True

File: frontend/src/models/match.js
import Competitor from './competitor';

export default class Match {
  constructor(id, ordinal, competitor1, competitor2, competitor1Score, competitor2Score,
    nextMatchId, status) {
    this.id = id;
    this.ordinal = ordinal;
    this.competitor1 = competitor1;
    this.competitor2 = competitor2;
    this.competitor1Score = competitor1Score;
    this.competitor2Score = competitor2Score;
    this.nextMatchId = nextMatchId;
    this.status = status;
  }

  static fromCreateResponseBody(response) {
    return new Match(
      response.id,
      response.ordinal,
      Competitor.fromCreateResponseBody(response.competitor_1),
      Competitor.fromCreateResponseBody(response.competitor_2),
      response.competitor_1_score,
      response.competitor_2_score,
      response.next_match_id,
      response.status,
    );
  }
}
